### ISSUE TITLE
Enumerable#filter, Enumerator::Lazy#filter are not aliases

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1610]
`Enumerable#filter`, `Enumerable#filter!` are aliases for `Enumerable#select`, `Enumerable#select!` since Ruby 2.6. See https://bugs.ruby-lang.org/issues/13784

To reproduce:

```ruby
(1..50).filter { |num| num % 2 == 0 }
```

Error:

```
Traceback (most recent call last):
        2: from ~/.rubies/truffleruby-1.0.0-nightly/jre/languages/ruby/bin/irb:31:in `<main>'
        1: from (irb):2
NoMethodError (undefined method `filter' for 1..50:Range)
```

**Comment by chrisseaton** [comment]
Is Enumerable#filter something you have a use-case for as well?

Just to be clear, we are aware of all new features. We tag then as not working yet in our specs https://github.com/oracle/truffleruby/blob/f6b387c4e239dec178094b4247cf4a3c5bc3b04f/spec/tags/core/enumerable/filter_tags.txt and it's often us who writes the specs. So you don't need to spend your time looking up MRI issues to go with your reports - we can find that documentation.

But we are definitely interested in missing features which you need for something - in that case we will put them to the front of the queue.

**Comment by deepj** [comment]
In fact, I prefer to use `#filter` since Ruby 2.6. And I use it’s in a small sceipt importing CSV where I need filter out some useless data. Also the reason is described in the issue, which is same like me, is compability with other programming languages. I use occasinaly JavaScript. So I decided to use it from Ruby 2.6.

**Comment by chrisseaton** [comment]
Yes I often write `filter` first time myself and have to rethink to write `select` each time!

**Comment by nirvdrum** [comment]
As far as I can tell, we're missing `Enumerable#filter` and `Enumerator#filter`. Neither `Enumerable` nor `Enumerator` have a `filter!` method, although `Array` and `Hash` do.

**Comment by nirvdrum** [comment]
The missing aliases have been implemented in 59a79086eec7ddbcab3b1f100c8e41f4b18c8d70. Unfortunately, I missed the code freeze for the next release (20.0.0-beta1) so this fix will become available in the release after that (20.0.0-beta2).
